engine/src/main/java/io/camunda/zeebe/engine/processing/distribution/CommandRedistributor.java

```diff
@@ -9,35 +9,66 @@
 
 import static io.camunda.zeebe.protocol.Protocol.DEPLOYMENT_PARTITION;
 
-import io.camunda.zeebe.engine.state.immutable.DeploymentState;
-import io.camunda.zeebe.protocol.impl.record.value.deployment.DeploymentRecord;
+import io.camunda.zeebe.engine.state.immutable.DistributionState;
+import io.camunda.zeebe.protocol.impl.record.value.distribution.CommandDistributionRecord;
+import io.camunda.zeebe.stream.api.InterPartitionCommandSender;
 import io.camunda.zeebe.stream.api.ReadonlyStreamProcessorContext;
 import io.camunda.zeebe.stream.api.StreamProcessorLifecycleAware;
 import java.time.Duration;
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.Map;
 import java.util.function.Predicate;
-import org.agrona.DirectBuffer;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class CommandRedistributor implements StreamProcessorLifecycleAware {
+/**
+ * The Command Redistributor provides a mechanism to retry sending {@link
+ * CommandDistributionRecord}s to other partitions. This is needed because the communication between
+ * partitions is unreliable.
+ *
+ * <p>A simple exponential backoff is used for retrying these pending distributions. This
+ * exponential backoff is statically configured to start of at 10 seconds {@link
+ * #COMMAND_REDISTRIBUTION_INTERVAL} until it reaches a maximum of 5 minutes {@link
+ * #RETRY_MAX_BACKOFF_DURATION}, doubling every time. This backoff is tracked for each pending
+ * distribution individually.
+ */
+public final class CommandRedistributor implements StreamProcessorLifecycleAware {
```

Comment:
👍 Thanks for adding all the javadoc to the redistributor. Makes it a lot more easy to understand!